### PR TITLE
docs: Updated docs for 0.1.25 release

### PR DESCRIPTION
**Pull request by drew-viles**
## Change description
* Updated the docs with reference to the new `v0.1.25` release

## Related issues
N/A
## Additional context
N/A

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # Image Builder Releases
 
-The current release of Image Builder is [v0.1.24][] (February 21, 2024). The corresponding container image is `registry.k8s.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.24`.
+The current release of Image Builder is [v0.1.25][] (February 21, 2024). The corresponding container image is `registry.k8s.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.25`.
```
Comment (mboersma) [inline RELEASE.md:3]: We should change the date here to March 13.

File: docs/book/src/capi/releasing.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Image Builder Releases
 
-The current release of Image Builder is [v0.1.24][] (February 21, 2024). The corresponding container image is `registry.k8s.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.24`.
+The current release of Image Builder is [v0.1.25][] (February 21, 2024). The corresponding container image is `registry.k8s.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.25`.
```
Comment (mboersma) [inline docs/book/src/capi/releasing.md:3]: Same here: change the date.

File: RELEASE.md (hunk 1 above)
Comment (drew-viles) [reply RELEASE.md:3]: Good spot! I used the auto doc creator :-p some tweaks needed. I'll manually do it for now and make a note of it

**Review by mboersma — APPROVED** [review]
/lgtm

**Comment by drew-viles** [comment]
Thanks for the eagle eyes @mboersma - I'm usually the one who has them tbf but I've been slightly reduced on the sleep side so missed that date! Updated now.

File: docs/book/src/capi/releasing.md [hunk 3]
```diff
@@ -33,12 +33,13 @@ Pushing the tag in the previous step triggered a job to build the container imag
 - Images are built by the [post-image-builder-push-images][] job. This will push the image to a [staging repository][].
 - Wait for the above post-image-builder-push-images job to complete and for the tagged image to exist in the staging directory.
 - If you don't have a GitHub token, create one via [Personal access tokens][]. Make sure you give the token the `repo` scope.
+- Make sure you have a clone of [k8s.io](https://github.com/kubernetes/k8s.io) otherwise the next step will not work.
```
Comment (AverageMarcus) [inline docs/book/src/capi/releasing.md:36]: 👍  And this is why it's good to have new people run through these 😁

**Review by AverageMarcus — COMMENTED** [review]
/lgtm

**Comment by drew-viles** [comment]
All looks good and ready to go:
```
docker pull registry.k8s.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.25
v0.1.25: Pulling from scl-image-builder/cluster-node-image-builder-amd64
23828d760c7b: Pull complete
1f4690091e60: Pull complete
4f4fb700ef54: Pull complete
4bca955d70d5: Pull complete
1e30c9b01c32: Pull complete
f0b89f763ddd: Pull complete
087e8d3cf06f: Pull complete
be6a28b87896: Pull complete
ef148baa628d: Pull complete
21f043eb5768: Pull complete
9e743a1f01fc: Pull complete
Digest: sha256:dd7d2afe7155fb60095235c7b78efa05846ec2df3774190bd994bb238bfb0bf5
Status: Downloaded newer image for registry.k8s.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.25
registry.k8s.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.25
```

**Pull request merged**